src/apps/intel_mp/intel_mp.lua

```diff
@@ -824,25 +824,38 @@ end
 
 function Intel:sync_stats ()
    local set, stats = counter.set, self.stats
-   set(stats.speed, self:link_speed())
-   set(stats.status, self:link_status() and 1 or 2)
-   set(stats.promisc, self:promisc() and 1 or 2)
-   set(stats.rxbytes, self:rxbytes())
-   set(stats.rxpackets, self:rxpackets())
-   set(stats.rxmcast, self:rxmcast())
-   set(stats.rxbcast, self:rxbcast())
-   set(stats.rxdrop, self:rxdrop())
-   set(stats.rxerrors, self:rxerrors())
-   set(stats.txbytes, self:txbytes())
-   set(stats.txpackets, self:txpackets())
-   set(stats.txmcast, self:txmcast())
-   set(stats.txbcast, self:txbcast())
-   set(stats.txdrop, self:txdrop())
-   set(stats.txerrors, self:txerrors())
-   set(stats.rxdmapackets, self:rxdmapackets())
-   for idx = 1, #self.queue_stats, 2 do
-      local name, register = self.queue_stats[idx], self.queue_stats[idx+1]
-      set(stats[name], register())
+   if self.master then
+      set(stats.speed, self:link_speed())
```

Comment:
AFAIU, speed is not a counter that increments; can we define this one on all apps?

Reply:
Yep, np. I think other counters such as `status` and `promisc`, should behave in the same way.